sdk/modelsrepository/Azure.Iot.ModelsRepository/src/RepositoryHandler.cs

```diff
@@ -16,19 +17,22 @@ internal class RepositoryHandler
         private readonly IModelFetcher _modelFetcher;
         private readonly Guid _clientId;
         private readonly ClientDiagnostics _clientDiagnostics;
+        private readonly Uri _repositoryUri;
+        private readonly ResolverClientOptions _clientOptions;
 
-        public Uri RepositoryUri { get; }
-        public ResolverClientOptions ClientOptions { get; }
-
-        public RepositoryHandler(Uri repositoryUri, ClientDiagnostics clientdiagnostics, ResolverClientOptions options = null)
+        public RepositoryHandler(Uri repositoryUri, ClientDiagnostics clientdiagnostics, ResolverClientOptions options)
         {
-            ClientOptions = options ?? new ResolverClientOptions();
-            RepositoryUri = repositoryUri;
+            Argument.AssertNotNull(options, nameof(options));
```

Comment:
Options shouldn't be null and we will not create a new instance if it is, the ResolverClient ctor will take care of it.